plugins/scaffolder-backend-module-bitbucket-cloud/src/actions/bitbucketCloudPullRequest.ts

```diff
@@ -0,0 +1,464 @@
+  authorization: string;
+  apiBaseUrl: string;
+}) => {
+  const {
+    workspace,
+    repo,
+    title,
+    description,
+    targetBranch,
+    sourceBranch,
+    authorization,
+    apiBaseUrl,
+  } = opts;
+
+  let response: Response;
+  const data: RequestInit = {
+    method: 'POST',
+    body: JSON.stringify({
+      title: title,
+      summary: {
+        raw: description,
+      },
+      state: 'OPEN',
+      source: {
+        branch: {
+          name: sourceBranch,
+        },
+      },
+      destination: {
+        branch: {
+          name: targetBranch,
+        },
+      },
+    }),
+    headers: {
+      Authorization: authorization,
+      'Content-Type': 'application/json',
+    },
+  };
+
+  try {
+    response = await fetch(
+      `${apiBaseUrl}/repositories/${workspace}/${repo}/pullrequests`,
+      data,
+    );
+  } catch (e) {
+    throw new Error(`Unable to create pull-reqeusts, ${e}`);
+  }
+
+  if (response.status !== 201) {
+    throw new Error(
+      `Unable to create pull requests, ${response.status} ${
+        response.statusText
+      }, ${await response.text()}`,
+    );
+  }
+
+  const r = await response.json();
+  return r.links.self.href;
+};
+
+const findBranches = async (opts: {
+  workspace: string;
+  repo: string;
+  branchName: string;
+  authorization: string;
+  apiBaseUrl: string;
+}) => {
+  const { workspace, repo, branchName, authorization, apiBaseUrl } = opts;
+
+  let response: Response;
+  const options: RequestInit = {
+    method: 'GET',
+    headers: {
+      Authorization: authorization,
+      'Content-Type': 'application/json',
+    },
+  };
+
+  try {
+    response = await fetch(
+      `${apiBaseUrl}/repositories/${workspace}/${repo}/refs/branches?q=${encodeURIComponent(
+        `name = "${branchName}"`,
+      )}`,
+      options,
+    );
+  } catch (e) {
+    throw new Error(`Unable to get branches, ${e}`);
+  }
+
+  if (response.status !== 200) {
+    throw new Error(
+      `Unable to get branches, ${response.status} ${
+        response.statusText
+      }, ${await response.text()}`,
+    );
+  }
+
+  const r = await response.json();
+
+  return r.values[0];
+};
+const createBranch = async (opts: {
+  workspace: string;
+  repo: string;
+  branchName: string;
+  authorization: string;
+  apiBaseUrl: string;
+  startBranch: string;
+}) => {
+  const {
+    workspace,
+    repo,
+    branchName,
+    authorization,
+    apiBaseUrl,
+    startBranch,
+  } = opts;
+
+  let response: Response;
+  const options: RequestInit = {
+    method: 'POST',
+    body: JSON.stringify({
+      name: branchName,
+      target: {
+        hash: startBranch,
+      },
+    }),
+    headers: {
+      Authorization: authorization,
+      'Content-Type': 'application/json',
+    },
+  };
+
+  try {
+    response = await fetch(
+      `${apiBaseUrl}/repositories/${workspace}/${repo}/refs/branches`,
+      options,
+    );
+  } catch (e) {
+    throw new Error(`Unable to create branch, ${e}`);
```

Comment:
Is `e` here as `string` or `Error`?
If it is Error, it has to be treated differently to show the details of the error properly